src/ai/backend/storage/vfs/__init__.py

```diff
@@ -676,5 +684,8 @@ async def delete_files(
         for p in target_paths:
             if p.is_dir() and recursive:
                 await self.fsop_model.delete_tree(p)
-            else:
-                await aiofiles.os.remove(p)
+            elif p.is_file():
+                if self.watcher is not None:
+                    await self.watcher.request_task(DeletePathTask(p))
+                else:
+                    await aiofiles.os.remove(p)
```

Comment:
I believe there might be a better way to decide whether the watcher should monitor, rather than using if statements for branching.
At the very least, is there a way to wrap the file removal and write operations at the time of creation?